cvat/apps/webhooks/signals.py

```diff
@@ -165,30 +172,37 @@ def pre_save_resource_event(sender, instance, **kwargs):
 @receiver(post_save, sender=Invitation, dispatch_uid=__name__ + ":invitation:post_save")
 @receiver(post_save, sender=Membership, dispatch_uid=__name__ + ":membership:post_save")
 def post_save_resource_event(sender, instance, created, **kwargs):
-    resource_name = instance.__class__.__name__.lower()
+    selected_webhooks = instance._webhooks_selected_webhooks
+    del instance._webhooks_selected_webhooks
 
-    event_type = event_name("create" if created else "update", resource_name)
-    if event_type not in map(lambda a: a[0], EventTypeChoice.choices()):
+    if not selected_webhooks:
         return
 
-    filtered_webhooks = select_webhooks(instance, event_type)
-    if not filtered_webhooks:
-        return
+    resource_name = instance.__class__.__name__.lower()
+    event_type = event_name("create" if created else "update", resource_name)
+
+    serializer = get_serializer(instance=instance)
 
     data = {
         "event": event_type,
-        resource_name: get_serializer(instance=instance).data,
+        resource_name: serializer.data,
         "sender": get_sender(instance),
     }
 
-    if not created:
-        if before_update := getattr(instance, "_before_update", None):
-            data["before_update"] = before_update
-        else:
-            return
+    old_data = instance._webhooks_old_data
+    del instance._webhooks_old_data
+
+    if not created and old_data is not None:
+        diff = get_instance_diff(old_data=old_data, data=serializer.data)
+
+        if diff:
+            data["before_update"] = {
+                attr: value["old_value"]
+                for attr, value in diff.items()
+            }
```

Comment:
Probably it will be better to set `_webhooks_old_data` attribute only when object was updated and `_webhooks_selected_webhooks` is not empty
```suggestion
    if not created and (old_data := getattr(instance, '_webhooks_old_data', None)):
        del instance._webhooks_old_data

        if diff := get_instance_diff(old_data=old_data, data=serializer.data):
            data["before_update"] = {
                attr: value["old_value"]
                for attr, value in diff.items()
            }
```

Reply:
I thought through this code again, and I decided to update it in a different way. I want to make sure that the value of `created` is consistent between the `pre_save` and `post_save` handlers, so I now initialize it based on the value of `_webhooks_old_data`.